blog/2022-09-24-LFX-Blog-Carol.md

```diff
@@ -0,0 +1,47 @@
+
+## Understanding Code
+
+A large portion of my mentorship focused on understanding the errors that were printed out. There were several different kinds of errors, such as ones that dealt with the console input, console output, pipeline input, and more, which validated whether the input or output existed in the code and where it led to, another connector or the console. While this may seem simple, this ended up leading me into the rabbit hole of the source code as I traversed through numerous files and functions to determine where the error went through and what I could do to the preexisting code to read and perform functions onto the error.
+
+To do a lot of this, I learned the process of error logging. I defined each error there in great detail, such as what it is, what it does currently, what it’s supposed to do, and gradually updated it every time it was needed. Throughout implementing messages to these errors, I also learned more about version control with git. While I did have practice on the basics like git pull and git push, I also learned about git rebase, git merge, and more.
+
+## Error Handling
+
+Error handling was the main focus of the project. As my mentor suggested, I focused on one error at a time from the error logs. First, I focused on the worst case- a case where nothing was printed and the program ran as if everything was normal, except there was no output. This error focused on the fact that the output did not exist, so to fix the error, I wrote a separate function that checked whether the output existed, then connected it to the `ConnectInput` structs while sending an error statement and a status report to the system. This process introduced me to focusing on small incremental steps rather than solving the problem as a whole and allowed me to not get overwhelmed. Additionally, `outputs` had to be added to the `Executable Graph` struct, which led me to implementing outputs in related functions and building a hashmap for the representation of its graph. With that, the basic problem of catching the errors was solved for that error problem.
+
+After that led to defining what the error was, which involved adding a `port` struct to both `ConnectInput` and `ConnectOutput` to connect inputs. This port struct wasn’t needed before but with changes that were made later on, `port` proved to be very useful in defining what the port was to be able to be used in other code. Because this struct was added, changes then had to be made to all the functions that used this struct and a port had to be defined in those functions.
```

Comment:
```suggestion
After that led to defining what the error was, which involved adding a `port` struct to both `ConnectInput` and `ConnectOutput` to connect inputs. This port struct wasn’t needed before but with changes that were made later on, `port` proved to be very useful in defining what the port was to be able to be used in other code. Because this field was added to the struct, changes then had to be made to all the functions that used this struct and a port had to be defined in those functions.
```